.github/workflows/docker.yml

```diff
@@ -51,16 +51,10 @@ jobs:
       - name: Build client binary
         run: |
           cargo install --path linera-service --bin linera --bin linera-server --debug
-      - name: Build Docker image
-        run: docker build . -f docker/Dockerfile -t linera
       - name: Run Compose
         run: |
           cd docker
           ./compose.sh
       - name: Sync
-        uses: nick-fields/retry@v2
```

Comment:
why do you not need the retry any more?

Reply:
Before we were just running the script as a background process, so it exited immediately and went to the `Sync` step. That's why it would fail, because it would have to wait until we built the docker image, run the `linera` commands to generate the wallet and configs, and _then_ run `docker compose up` and wait for the services to be healthy. So it had to wait for all that to happen before the `Sync` could succeed, which is why we needed the retry with a generous delay.
Now that we'll only exit the `Run Compose` step when all those things are done and services are healthy (after the changes on #2683), retrying should not be needed anymore.

Reply:
I added the retry back just in case, to avoid the test from being flaky, but now it'll try the `sync-balance`, then wait 10 seconds, try again... until the 2 minutes timeout